### PR TITLE
Documentation for Suggested Config feature

### LINKED ISSUE #5821: Add documentation for the auto configuration feature
This relies on https://github.com/RasaHQ/rasa/issues/5820 being implemented.

Should probably also wait until we know more about how Rasa 2.0 is gonna look like, since many things are changing at the moment.

### PR DESCRIPTION
**Pull request by chkoss**
Explain the new Suggested Config feature in the documentation
Fixes #5821 

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/docs/core/policies.mdx [hunk 1]
```diff
@@ -25,6 +25,9 @@ to the desired number of maximum predictions.
 
 Your project’s `config.yml` file takes a `policies` key
 which you can use to customize the policies your assistant uses.
+You can also leave this key empty. Then the
```
Comment (erohmensing) [inline docs/docs/core/policies.mdx:28]: Might be good to show a little code snippet for each of these "left empty" as some users might take that as not including it at all. 

"You can also leave this key empty:
```

policies: 
# intentionally left empty

```
Then the suggested...

File: docs/docs/nlu/choosing-a-pipeline.mdx [hunk 2]
```diff
@@ -30,21 +30,15 @@ components you want to use directly in the configuration file. See
...
-```yaml (docs/sources/data/configs_for_docs/default_english_config.yml)
-```
-
-If your training data is not in English, start with the following pipeline:
-
-```yaml (docs/sources/data/configs_for_docs/default_config.yml)
-```
+To get started, you can let the
+[Suggested Config](../api/model-configuration.mdx#suggested-config) feature choose a
+default pipeline for you.
```
Comment (erohmensing) [inline docs/docs/nlu/choosing-a-pipeline.mdx:35]: Do you have to fill in the language parameter to do so? If so i'd specify that, and show an example: 
```
language: fr  # your 2-letter language code

pipeline:
# intentionally left empty
```

(i don't really know the code of how it chooses a pipeline for you or if it just chooses the language-agnostic one)

File: docs/docs/api/model-configuration.mdx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+The configuration file defines the NLU and Core components that your model will use. The
+`language` and `pipeline` keys specify how the NLU model should be built. You can
+learn about the different NLU pipelines [here](../nlu/choosing-a-pipeline.mdx). The
+`policies` key defines the [policies](../core/policies.mdx) that the Core model will
+use.
+
+## Suggested Config
+
+You always need to specify the `language` key in your `config.yml` file.
```
Comment (erohmensing) [inline docs/docs/api/model-configuration.mdx:17]:
```suggestion
You always need to specify the `language` key in your `config.yml` file with the 2-letter ISO language code.
```

File: docs/docs/core/policies.mdx (hunk 1 above)
Comment (erohmensing) [reply docs/docs/core/policies.mdx:28]: Nevermind I just read the actual documentation that says you can leave it out or empty -- I'd write that here

File: docs/docs/api/model-configuration.mdx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+The two other keys (`pipeline` and `policies`) can be left out or left empty. When running `rasa train`, the
+"Suggested Config" feature will select a default configuration for the missing key(s)
```
Comment (erohmensing) [inline docs/docs/api/model-configuration.mdx:20]:
```suggestion
Suggested Config feature will select a default configuration for the missing key(s)
```

File: docs/docs/nlu/choosing-a-pipeline.mdx (hunk 2 above)
Comment (chkoss) [reply docs/docs/nlu/choosing-a-pipeline.mdx:35]: I like the idea of providing examples. Will add one to the main documentation for this feature, too.

**Comment by wochinge** [comment]
re-ran the tests

File: rasa/nlu/components.py [hunk 5]
```diff
@@ -97,7 +97,7 @@ def validate_only_one_tokenizer_is_used(pipeline: List["Component"]) -> None:
 
     if len(tokenizer_names) > 1:
         raise InvalidConfigError(
-            f"More then one tokenizer is used: {tokenizer_names}. "
+            f"More than one tokenizer is used: {tokenizer_names}. "
```
Comment (wochinge) [inline rasa/nlu/components.py:100]: 🚀

File: tests/nlu/test_config.py [hunk 6]
```diff
@@ -159,10 +161,16 @@ def config_files_in(config_directory: Text):
     "config_file",
     config_files_in("data/configs_for_docs") + config_files_in("docker/configs"),
 )
-def test_train_docker_and_docs_configs(config_file: Text):
+async def test_train_docker_and_docs_configs(
+    config_file: Text, monkeypatch: MonkeyPatch
+):
     content = io_utils.read_yaml_file(config_file)
```
Comment (wochinge) [inline tests/nlu/test_config.py:168]:
```suggestion
```

File: tests/nlu/test_config.py [hunk 7]
```diff
@@ -159,10 +161,16 @@ def config_files_in(config_directory: Text):
...
 
-    loaded_config = config.load(config_file)
+    monkeypatch.setattr(autoconfig, "_dump_config", Mock())
+    importer = RasaFileImporter(config_file=config_file)
+    imported_config = await importer.get_config()
+
+    loaded_config = config.load(imported_config)
 
     assert len(loaded_config.component_names) > 1
     assert loaded_config.language == content["language"]
```
Comment (wochinge) [inline tests/nlu/test_config.py:176]: ?
```suggestion
    assert loaded_config.language == imported_config["language"]
```

**Pull request merged**